src/api/API.ts

```diff
@@ -38,7 +38,13 @@ export class API {
 	 * @param path the vault relative path of the file to import
 	 */
 	public async importJs(path: string): Promise<any> {
-		const fullPath = this.app.vault.adapter.getResourcePath(path);
+		let fullPath = this.app.vault.adapter.getResourcePath(path);
+		if (!fullPath.includes('?')) {
```

Comment:
What if the path on mobile includes a `?` somewhere along the way?

Reply:
Something like `this/is/a?path/foo.js`

Reply:
Can it have one? Isn't `?` an illegal character to have in a path or in a filename? If you're talking about a mobile path that already contains query parameters then It shouldn't be the case since on mobile, `getResourcePath` doesn't append anything at the end of the full path

Reply:
https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/os/FileUtils.java;l=1008?q=isValidFatFilenameChar
Seems like you are correct and `?` as an invalid character for file paths.